src/libraries/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/Pbkdf2Implementation.Windows.cs

```diff
@@ -0,0 +1,270 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+using BCryptAlgPseudoHandle = Interop.BCrypt.BCryptAlgPseudoHandle;
+using BCryptBuffer = Interop.BCrypt.BCryptBuffer;
+using BCryptOpenAlgorithmProviderFlags = Interop.BCrypt.BCryptOpenAlgorithmProviderFlags;
+using NCryptBCryptBufferDescriptors = Interop.BCrypt.NCryptBCryptBufferDescriptors;
+using NTSTATUS = Interop.BCrypt.NTSTATUS;
+
+namespace Internal.Cryptography
+{
+    internal partial class Pbkdf2Implementation
+    {
+        private static readonly bool s_usePseudoHandles = OperatingSystem.IsWindowsVersionAtLeast(10, 0, 0);
+
+        // For Windows 7 we will use BCryptDeriveKeyPBKDF2. For Windows 8+ we will use BCryptKeyDerivation
+        // since it has better performance.
+        private static readonly bool s_useKeyDerivation = OperatingSystem.IsWindowsVersionAtLeast(8, 0, 0);
```

Comment:
Does anyone run Win8?  I'm probably fine with just having Win7+ and Win10+ code, and skipping that 8 could be slightly better.  Especially if it simplifies things.  (If it's just this one piece of detection, fine.  But if we have "older API, newer API with SafeHandles, newer API with PseudoHandles" I'd like to eliminate that middle step of complexity)

Reply:
> and skipping that 8 could be slightly better

@bartonjs the difference between `BCryptKeyDerivation` and `BCryptDeriveKeyPBKDF2` is significant, 1.5x to 2x faster with the former, with common workload parameters. While I would be happy to simplify, it's not a handleful of microseconds. If we wanted to have this so that `DataProtection` could switch, to something in-box, Windows Server 2012 (R2) would take a non-trivial hit.

Reply:
Sad face.  OK.